src/main/java/uk/gov/hmcts/reform/roleassignment/domain/service/deleteroles/DeleteRoleAssignmentOrchestrator.java

```diff
@@ -281,5 +291,83 @@ private void updateRequestStatus(AssignmentRequest assignmentRequest, Status sta
 
     }
 
+    @SuppressWarnings("unchecked")
+    public ResponseEntity<Void> deleteRoleAssignmentByQuery(MultipleQueryRequest multipleQueryRequest) {
+
+        //1. create the request Object
+        if (CollectionUtils.isNotEmpty(multipleQueryRequest.getQueryRequests())) {
+            request = parseRequestService.prepareDeleteRequest("", "", "", "");
+            request.setLog(DELETE_BY_QUERY);
+            assignmentRequest = new AssignmentRequest(request, Collections.emptyList());
+        } else {
+            throw new BadRequestException(V1.Error.BAD_REQUEST_MISSING_PARAMETERS);
+
+        }
+
+        //2. Call persistence service to store only the request
+        persistInitialRequestForDelete();
+
+        //3. retrieve all assignment records by multiple query
+        List<Assignment> requestedRoles = fetchRoleAssignmentsByMultipleQuery(multipleQueryRequest);
+        var responseHeaders = new HttpHeaders();
+        responseHeaders.add(
+            "Total-Records",
+            Long.toString(persistenceService.getTotalRecords())
+        );
+
+        if (requestedRoles.isEmpty()) {
+            requestEntity.setStatus(Status.APPROVED.toString());
+            persistenceService.updateRequest(requestEntity);
+            return new ResponseEntity<>(responseHeaders, HttpStatus.OK);
+        } else {
+            //update the records status from Live to Delete_requested for drool to approve it.
+            requestedRoles.stream().forEach(roleAssignment -> roleAssignment.setStatus(Status.DELETE_REQUESTED));
+        }
+
+        //Perform others delete steps
+        ResponseEntity<Void> responseEntity = performOtherStepsForDelete(
+            "",
+            (List<RoleAssignment>) (List<?>) requestedRoles
+        );
+        if (responseEntity.getStatusCode() == HttpStatus.NO_CONTENT) {
+            return  new ResponseEntity<>(responseHeaders, HttpStatus.OK);
+        }
+
+        return responseEntity;
+
+    }
+
+    private List<Assignment> fetchRoleAssignmentsByMultipleQuery(MultipleQueryRequest multipleQueryRequest) {
+        List<List<Assignment>> assignmentRecords = new ArrayList<>();
+        assignmentRecords.add(persistenceService.retrieveRoleAssignmentsByMultipleQueryRequest(
+            multipleQueryRequest,
+            0,
+            0,
```

Comment:
This will delete only first 20 records selected by all the query requests, what if total records selected more than 20?

Reply:
We have iteration logic base on number of  total records on select query than accumulate the final list.